### ISSUE TITLE
Observer Block Malfunctioning

### ISSUE THREAD

**Issue by cheshire1111** [Roadhog360/Et-Futurum-Requiem#276]
### Please check any boxes that apply to you and your issue

- [X] I use a translator application to post this issue.

- [ ] This is a crash. Please upload, Pastebin, Gist or copypaste the whole crash report along with this issue.

- [ ] This is a mod incompatibility. If I do this in vanilla Forge with only Et Futurum Requiem installed, it works normally.

### Version number of Et-Futurum-Requiem (IMPORTANT)
Et Futurum Requiem 2.4.6

### Describe the issue (IMPORTANT)
Hi, I'm having issues with the observer block in the Et Futurum Requiem mod for Minecraft. The block seems to activate on its own, even without a redstone signal, and it's not possible to use it for certain mechanics, such as redstone clocks or observer-based farms. I apologize if I'm mistaken about anything, as I'm not very familiar with redstone, but it seems like there might be an error, since I followed all the steps in a tutorial on how to use the observer block, and it didn't work correctly. Please let me know if there's something I missed, and thank you for your help!

The tutorial I was trying to follow is this one: https://www.youtube.com/watch?v=ZrmqFMWXIDw&t=165s
I simply can't create this structure without an observer activating, so I'm unable to do anything similar.

**Comment by Roadhog360** [comment]
I don't understand the problem, wouldn't placing all those blocks cause them to activate each other? That's just what it's supposed to do. Observers also do not require a redstone signal to observe blocks. So you say "even without a redstone signal" but that's just intended.

Can you please elaborate on exactly what is not working? Which step is failing?

**Comment by cheshire1111** [comment]
> Não entendo o problema, colocar todos esses blocos não faria com que eles ativassem uns aos outros? Isso é exatamente o que deveria fazer. Os observadores também não requerem um sinal de redstone para observar blocos. Então você diz "mesmo sem um sinal de redstone", mas isso é apenas intencional.
> 
> Você pode detalhar exatamente o que não está funcionando? Qual etapa está falhando?

Of course, I'm sorry. I'm not really familiar with redstone, I'm actually still learning and I'm used to playing on version 1.7.10 where there were no observers. But I think something might be wrong, because I'm simply unable to follow tutorials that involve this block. If you take a look at the video, you'll see what I'm trying to do. When I'm finishing the first layer of the structure, the left side lights up for some reason, preventing me from making the front structure correctly (it should be unlit as in the video, where the observers point forward), so I can light it up later with repeaters and make the redstone lamp "disco". I'm sorry if I'm being ignorant about something, I'm just really confused XD. Thank you so much for the quick response, I really appreciate being well taken care of.

**Comment by cheshire1111** [comment]
So in my game, the observers with the exact same structure (I made sure no block was out of place) stay lit up like this: https://imgur.com/a/EWeyr7Y 
Which results in all the lamps being lit up, preventing me from doing the light dance and making the redstone system behind it work.

When I imagine that it should actually be like this (according to the tutorial videos, of course. I might be missing something important and I would appreciate if you could point out the problem if you know it): https://imgur.com/a/jEaZwxH

**Comment by GregoriusT** [comment]
Hrrm, can you put Lamps adjacent to the wall if observers behind this? I wonder if an errand block update is caused somewhere around that area. Especially if you have two observers facing each other, which would result in an infinite Redstone Clock.

@Roadhog360 quick question, you made sure observers only accept Block updates from their Face right? Not from any of their other Sides.

**Comment by Roadhog360** [comment]
I did not design observers, @embeddedt did. I can actually do some testing now though.

@cheshire1111 just to be sure, can you try replicating this build in 1.19 just to be 100% sure you are building it right?

**Comment by DarianLStephens** [comment]
The only problem I've seen with observers is them not activating after being moved by a piston, which causes some big differences in redstone possibilities.

**Comment by Roadhog360** [comment]
I have narrowed down the issue.

![image](https://github.com/Roadhog360/Et-Futurum-Requiem/assets/11903511/789bce2f-9682-49db-a7c4-a684a1ab00f7)

Observers appear to observe blocks being powered by redstone updates, in vanilla 1.19 they do not. This causes the observers within the lighting contraption to indefinitely observe each other's redstone updates.

**Comment by Roadhog360** [comment]
No, they do not, otherwise explain why in vanilla the powered cobblestone did not trigger the back observer to update? Yes it is right, I built the exact same contraption in both versions. Update the front of observer with any block, it will not fire the note block.

**Comment by GregoriusT** [comment]
Weird, pretty sure I saw contraptions make use of having blocks between observers. Must've misunderstood something then.

**Comment by cheshire1111** [comment]
so what should i understand? please note that english is not my native language so there is a bit more difficulty in communication. Yes, the lights worked in 1.19. the video tutorial is in a post 1.14 version. I rebuilt the structure 3 times to make sure it was correct, and unfortunately only on 1.7 it's not working. A point to highlight is that there were no redstones activating the structures of observers, in fact they activated themselves by themselves without any trace of redstone around. Then I added redstone to see if it had any effect, but then removed it. Anyway, what should I do? Is this really a problem that might be fixed in the future? (Don't get me wrong, I'm not in a hurry, I understand that mods take time to produce and fix) In any case, thanks for the quick response. You are doing a great job with the mod. I'm still playing 1.7.10 to complete some mods from there, and your mod brings good experiences.

**Comment by Roadhog360** [comment]
Allow me to explain in more detail. When a block is directly connected by a redstone component such as being above a redstone torch, the block is "powered". In modern versions where the observer is added, observers will not detect this change and therefore will not send a redstone signal in response.

What you are trying to build works by a chain of observers detecting each other's observer block updates. They're seeing the observer changed, not the redstone signal itself. In the Et Futurum Requiem observer, it's wrongly detecting that redstone pulse so every time one powers the other, the four observers adjacent to it wrongly detect this redstone pulse and will update, causing it to infinitely flash.

**Comment by cheshire1111** [comment]
oh yes! I think I understand what's going on after all. do you have plans to release a patch for this? thank you for your support! ♥️

**Comment by Roadhog360** [comment]
We unfortunately do not know how to fix this issue at the moment.

**Comment by OvermindDL1** [comment]
Does vanilla MC even have a concept of 'why' a neighbor update happened?  Is it remotely possible to get this information in a mod-compatible way?  Sure you could mixin to vanilla, but not every single mod that touches redstone for example...

**Comment by Roadhog360** [comment]
Who knows?
